.github/workflows/release.yml

```diff
@@ -0,0 +1,96 @@
+name: Release
+
+on:
+  push:
+    tags:
+      - "*"
+    branches:
+      - "main"
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  unix:
+    name: unix build
+
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-20.04
+            compiler: gcc
+          - os: ubuntu-20.04
+            compiler: clang
+          - os: macos-10.15
+            compiler: gcc
+          - os: macos-10.15
+            compiler: clang
```

Comment:
do we want to bump os to `ubuntu-22.04` and `macos-12`? also applies to other matrixes

Reply:
will try it out and test it.

Reply:
done